VAMobile/android/fastlane/Fastfile

```diff
@@ -202,17 +203,28 @@ platform :android do
 
   def promote_pre_prod
     begin
-      prod = google_play_track_version_codes(track:"production")
-      stash =  google_play_track_version_codes(track:"Pre-Prod Stash")
-      # if prod.max is greater or equal, promoting will fail because prod has a more recent build.
+      prod = google_play_track_version_codes(track: "production")
+      stash = google_play_track_version_codes(track: "Pre-Prod Stash")
+      # Check if promotion is allowed based on version codes
       if stash.max > prod.max
+        # Promote the version from Pre-Prod Stash to production
         supply(track: "Pre-Prod Stash", track_promote_to: "production", version_code: stash.max)
-        version = google_play_track_release_names(track:"production")
-        slack_release_success(version.max_by{ |s| Gem::Version.new(s) }, "Android")
+        # Delay briefly for 5 min to allow Google Play to process the update (optional, tweak as needed)
+        sleep 300
```

Comment:
In your testing, did the code resume properly after `sleep`?

Reply:
I didn't use the sleep for testing!

Reply:
I'm hesitant to approve this without having tested it because typically long wait intervals like that can be unreliable. You could implement a dynamic wait with a while loop that will retry every minute for 5 minutes like this:

```ruby
max_attempts = 5
interval = 60
attempts = 0

while attempts < max_attempts
  attempts += 1
  puts "Checking if release is live. Attempts: #{attempts}..."

  # Check if the promoted version is live on production
  live_version = google_play_track_version_codes(track: "production").max
  if live_version == stash.max
    # Get the release names and verify the promotion
    promoted_version = google_play_track_release_names(track: "production").max_by { |s| Gem::Version.new(s) }
    slack_release_success(promoted_version, "Android")
    puts "Release #{live_version} is live after checking #{attempts} time(s)."
    break
  end

  if attempts < max_attempts
    puts "Latest release version is not live yet. Waiting #{interval} seconds before retrying..."
    sleep(interval)
  else
    puts "Latest version is not live after #{max_attempts} attempts."
    raise "Latest version is not live after the allowed attempts"
  end
end
```
You could do some testing in the test lane to verify that the loop works.

Reply:
Tested with the delay and it resumes properly. 
<img width="976" alt="Screenshot 2024-12-16 at 2 20 08 PM" src="https://github.com/user-attachments/assets/acd257f1-552a-401f-8473-8b899001f62a" />

<img width="924" alt="Screenshot 2024-12-16 at 2 20 26 PM" src="https://github.com/user-attachments/assets/d58ddb51-e055-4662-8837-1cd92f9ba89b" />